wcomponents-core/src/main/java/com/github/bordertech/wcomponents/subordinate/AbstractCompare.java

```diff
@@ -107,7 +107,7 @@ protected boolean execute() {
 	@Override
 	protected boolean execute(final Request request) {
 		// Disabled triggers are always false
-		if ((trigger instanceof Disableable) && ((Disableable) trigger).isDisabled()) {
+		if ((trigger instanceof Disableable) && ((Disableable) trigger).isDisabled() && !(trigger instanceof Input && ((Input) trigger).isReadOnly())) {

```

Comment:
This is good. `Readonly` takes precedence over `Disabled`. This applies to the `Trigger`, but we should also make sure the server side logic is the same as the client for `actions`.

Should `mandatory/optional`, `hide/show`, `enable/disable` actions not be applied to `targets` that are `readonly` or `disabled` (except for the enable/disable of course). What does the client do for these actions?